Faster estimation of Lempel-Ziv Complexity

pranaysy opened the pull request:
#### Addresses speed issues highlighted in #14 
- Implementation maintained as per [Naereen's original code](https://github.com/Naereen/Lempel-Ziv_Complexity/commit/5537546dfcba37dae1079aeb8f8b57d68171ee7e) with some restructuring
- Switched from strings to integer arrays for speed gains using Numba
- Maps strings to their UTF-8 representations as 32-bit unsigned integers
- Switched to `np.bincount` instead of `set` for getting unique values faster
- Passes tests
#### Rough performance estimates
 - Hardware: HP Laptop with Intel Core-i5 1035G4 4C8T, 20GB RAM
 - Operating System: Pop OS 20.10 on kernel 5.8.0-7642-generic
 - Python Environment: Python 3.8.5, NumPy 1.19.2, Numba 0.51.2
![image](https://user-images.githubusercontent.com/8765418/112433166-df27af80-8d67-11eb-8b5a-e77246a1698f.png)

raphaelvallat commented:
Hi @pranaysy,

This is fantastic, thank you! A few comments:

1) This project is now deprecated and replaced by the pip-installable [AntroPy](https://github.com/raphaelvallat/antropy). Would you be able to submit the PR there as well? (I'll merge it in both projects)

2) Could you also add the changes (with link to the PR) in the [docs/changelog.rst](https://github.com/raphaelvallat/entropy/blob/master/docs/changelog.rst) file (under version 0.1.4)? No need to compile the full documentation, I'll do this when I release a new version.

Thanks again for all the hard work on this!

pranaysy commented:
> Hi @pranaysy,
> 
> This is fantastic, thank you! A few comments:
> 
>     1. This project is now deprecated and replaced by the pip-installable [AntroPy](https://github.com/raphaelvallat/antropy). Would you be able to submit the PR there as well? (I'll merge it in both projects)
> 
>     2. Could you also add the changes (with link to the PR) in the [docs/changelog.rst](https://github.com/raphaelvallat/entropy/blob/master/docs/changelog.rst) file (under version 0.1.4)? No need to compile the full documentation, I'll do this when I release a new version.
> 
> 
> Thanks again for all the hard work on this!

Thanks Raphael!
I have updated the changelog as you suggested.
I have submitted a PR to the [`antropy` repository](https://github.com/raphaelvallat/antropy/pull/1) as well, but I have not updated its changelog.

You're most welcome! :)

raphaelvallat commented:
Merging now! Thanks again!
